### PR TITLE
fix(jsii): symbolid for single-valued enums is incorrect

### PR DESCRIPTION
**Pull request by rix0rrr**
We used to be doing our own `symbolId` calculations, but this was silly:
TS has a perfectly cromulent function built-in to do the same.

Implement ours in terms of theirs.

This uncovered a bug in calculating symbolIds for single-valued enums, which is rectified as well.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: packages/jsii/lib/symbol-id.ts [hunk 1]
```diff
@@ -11,41 +39,21 @@ export function symbolIdentifier(
...
-  if (!namespace) {
+  const [, fileName, typeName] = groups; // typeName may be absent
+
+  const relFile = assemblyRelativeSourceFile(fileName);
+  if (!relFile) {
     return undefined;
   }
 
-  return `${namespace}:${inFileNameParts.join('.')}`;
+  return `${relFile}:${typeName ?? ''}`;
```
Comment (otaviomacedo) [inline packages/jsii/lib/symbol-id.ts:56]: Is it intentional that we leave the ':' in, even when there is no type name?

File: packages/jsii/lib/symbol-id.ts [hunk 2]
```diff
@@ -11,41 +39,21 @@ export function symbolIdentifier(
...
-        inFileNameParts.unshift(declSym.name);
-      }
-    }
-    decl = decl.parent;
-  }
-  if (!decl) {
+  const tsName = typeChecker.getFullyQualifiedName(sym);
+  // TypeScript fqn looks like "/path/to/file"[.name.in.file]
+  const groups = /^"([^"]+)"(?:\.(.*))?$/.exec(tsName);
+  if (!groups) {
```
Comment (otaviomacedo) [inline packages/jsii/lib/symbol-id.ts:45]: What does it mean if the name doesn't match the regex? Should we log a warning?

Comment (rix0rrr) [reply packages/jsii/lib/symbol-id.ts:45]: I don't know that we have the mechanisms to log a warning. I am fine ignoring this for now, we have end-to-end tests that will ensure everything works correctly. [Alarm on symptoms, not causes](https://paulbellamy.com/2017/08/symptoms-not-causes)

Comment (otaviomacedo) [reply packages/jsii/lib/symbol-id.ts:45]: I believe we do (`log4js`). But anyway, this is just a minor point.

**Review by otaviomacedo — APPROVED** [review]

**Pull request merged**